### DISCUSSION TITLE
Move files between libraries

### DISCUSSION THREAD

**Discussion by B1Z1** [nrwl/nx#10597] [Feature Requests]
Is there any way to move one file between libraries with auto importing? Because I have a file that is used in 100 libraries and my IDE cannot handle it automatically

**Comment by MetaBenji** [comment]
@B1Z1 did you ever find a solution for this?

**Reply by B1Z1** [reply]
Hi! We created a groupable library that exports everything from our core libraries. So if I need to move a file from one core library to another core library I can do it easily, because there's no need to change the import path inside the whole project.

Example: 

The core library has '@project/core' path.
Also, the core library exports in index.ts every library which are inside the core library.

Every library outside '@project/core', imports all files from '@project/core', instead of '@project/core/path/to/library'.

So the problem is partly solved :D. If you need example project, I can create one for you

**Comment by LayZeeDK** [comment]
We released @nxworker/workspace:move-file to solve this for all JS/TS Nx workspaces.

nx add @nxworker/workspace

nx generate @nxworker/workspace:move-file file/path.ts,or/glob*pattern.ts —project=target-project

https://www.npmjs.com/package/@nxworker/workspace
